Lazy/ILazy.cs

```diff
@@ -0,0 +1,19 @@
+﻿// MIT License
+// Copyright (c) 2024 Artem-Nesterenko2005
+// All rights reserved
+
+namespace Lazy
+{
+    /// <summary>
+    /// interface for lazy calculation.
+    /// </summary>
+    /// <typeparam name="T">the data type for lazy calculation.</typeparam>
+    public interface ILazy<T>

```

Comment:
```suggestion
    public interface ILazy<out T>
```
Если есть возможность указать вариантность, стоит ею пользоваться